utils/buildutils.go

```
@@ -0,0 +1,423 @@
+	dirPath, err := getBuildDir(buildName, buildNumber, projectKey, buildsDirPath)
+	if err != nil {
+		return err
+	}
+	log.Debug("Creating temp build file at: " + dirPath)
+	tempFile, err := ioutil.TempFile(dirPath, "temp")
+	if err != nil {
+		return err
+	}
+	defer tempFile.Close()
+	_, err = tempFile.Write([]byte(content.String()))
+	return err
+}
+
+func SaveBuildGeneralDetails(buildName, buildNumber, projectKey, buildsDirPath string, log Log) error {
+	partialsBuildDir, err := getPartialsBuildDir(buildName, buildNumber, projectKey, buildsDirPath)
+	if err != nil {
+		return err
+	}
+	log.Debug("Saving build general details at: " + partialsBuildDir)
+	detailsFilePath := filepath.Join(partialsBuildDir, BuildInfoDetails)
+	var exists bool
+	exists, err = isFileExists(detailsFilePath)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return nil
+	}
+	meta := buildinfo.General{
+		Timestamp: time.Now(),
+	}
+	b, err := json.Marshal(&meta)
+	if err != nil {
+		return err
+	}
+	var content bytes.Buffer
+	err = json.Indent(&content, b, "", "  ")
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(detailsFilePath, []byte(content.String()), 0600)
+	return err
+}
+
+// SavePartialBuildInfo saves the given partial in the builds directory.
+// The partial's Timestamp field is filled inside this function.
+func SavePartialBuildInfo(buildName, buildNumber, projectKey, buildsDirPath string, partial *buildinfo.Partial, log Log) error {
+	partial.Timestamp = time.Now().UnixNano() / int64(time.Millisecond)
+	b, err := json.Marshal(&partial)
+	if err != nil {
+		return err
+	}
+	var content bytes.Buffer
+	err = json.Indent(&content, b, "", "  ")
+	if err != nil {
+		return err
+	}
+	dirPath, err := getPartialsBuildDir(buildName, buildNumber, projectKey, buildsDirPath)
+	if err != nil {
+		return err
+	}
+	log.Debug("Creating temp build file at:", dirPath)
+	tempFile, err := ioutil.TempFile(dirPath, "temp")
+	if err != nil {
+		return err
+	}
+	defer tempFile.Close()
```

Let's not ignore the error returned from this method.